src/org/zaproxy/zap/extension/ascanrulesBeta/Csrftokenscan.java

```diff
@@ -191,10 +194,29 @@ public void scan() {
 					}
 					
 				}
-				// We clean up the cookies and perform again the request
-				// TODO wont this loose our session if we are authenticated?
+
+				// We keep only the "flagged as session" cookies, and perform again the request
+				// Get HTTP session names from config
+				OptionsParam options = Model.getSingleton().getOptionsParam();
+				HttpSessionsParam sessionOptions = options.getParamSet(HttpSessionsParam.class);
+				List<String> sessionIds = sessionOptions.getDefaultTokensEnabled();

```

Comment:
Ideally we should check that `sessionOptions` is non-null before using it (it can be null if the extension is disabled).